### DISCUSSION TITLE
`magit-section.el` does not render visibility indicator for sections with heading of length 1 and non-trivial content

### DISCUSSION THREAD

**Discussion by ncalexan** [magit/magit#4637] [Ideas]
There's a small issue where `magit-section.el` does not render a visibility indicator for a section with a heading of length exactly 1 and non-trivial content (so that there should be an indicator).

## What behavior you expected

The appropriate indicator to be displayed.

## What behavior you observed

No indicator for sections with headings `a` and `b`, like:

<img width="211" alt="magit-no-visibility-indicator" src="https://user-images.githubusercontent.com/1097182/158078573-333e0733-188c-4800-9411-9859a01682f3.png">

## How we can reproduce the issue

I think this can be arranged with a little elisp, but I don't have it to hand.  If it's needed I can produce it.

## magit-version

Nothing has changed here since at least 2.90, but I updated just to be sure:

```
M-x magit-version RET
Magit v3.3.0-208-g1b249d4b, Git 2.35.1, Emacs 28.0.60, darwin
```

I'm mostly just capturing this so that other consumers of `magit-section.el` might find it.  Given the comments around https://github.com/magit/magit/blob/1b249d4b3dc51b99840e4eb78ba22d9f7d15eee7/lisp/magit-section.el#L1454-L1458 I think this may just be part of the tradeoffs made for the visibility indicators generally, but perhaps it's never been noticed and there is a way to address it.  Fingers crossed!

**Comment by tarsius** [comment]
> No indicator for sections with headings a and b, like:

I am confused. How does one get Magit to produce such headings? As far as I can tell it doesn't use "Changes" as the heading of any section and it also does not show diff headings such as `---a/a` and `+++b/b`.

**Comment by ncalexan** [comment]
> > No indicator for sections with headings a and b, like:
> 
> I am confused. How does one get Magit to produce such headings? As far as I can tell it doesn't use "Changes" as the heading of any section and it also does not show diff headings such as `---a/a` and `+++b/b`.

This is specific to third-party consumers of `magit-section.el`.  I'm not aware of any way to get Magit to produce such headings, since in diffs, files are prefixed with `modified|added|deleted`, etc.  (The screenshot posted is from some WIP that I have of converting https://github.com/ananthakumaran/monky, a magit-alike for Mercurial, to use `magit-section.el`.)

**Reply by tarsius** [reply]
I'll have to setup some test code to check if the current kludge is still necessary.

Good luck updating Monkey!

**Comment by tarsius** [comment]
I've removed the kludge and doubt it ever really was necessary.

Just in case I have to look at this again; here's some code that might come in handy:

```elisp
(defun testibus ()
  (interactive)
  (with-current-buffer (get-buffer-create "testibus")
    (magit-mode)
    (setq buffer-read-only t)
    (let ((inhibit-read-only t))
      (erase-buffer)
      (save-excursion
        (magit-insert-section (outer)
          (magit-insert-section (a nil t)
            (magit-insert-heading "a")
            (magit-insert-section-body
              (insert "aaa\n")))
          (magit-insert-section (b)
            (magit-insert-heading "b")
            (insert "bbb\n")))))
    (pop-to-buffer (current-buffer))))
```
